test/test_lazybuffer.py

```diff
@@ -57,5 +58,14 @@ def test_shrink_const_then_cast(self):
     c = a.cat(b, dim=1)
     np.testing.assert_allclose(c.numpy(), np.concatenate((a.numpy(), b.numpy()), axis=1))
 
+  def test_const_dtype(self):
+    lb: LazyBuffer = Tensor([1], dtype=dtypes.int).lazydata
+    assert lb.const(1).base.arg == 1
+    assert type(lb.const(1).base.arg) == int
+
+    lb: LazyBuffer = Tensor([1], dtype=dtypes.float).lazydata
+    assert lb.const(1).base.arg == 1.0
+    assert type(lb.const(1).base.arg) == float
```

Comment:
this fails in master:
```
    assert type(lb.const(1).base.arg) == float
AssertionError
```

Reply:
it'd be very tedious to pick 1.0 vs 1 in mlops itself, i think LazyBuffer should just default to whatever the lb's dtype already is